### PR TITLE
Fix NPE in BzlmodRepoRuleFunction

### PR DESCRIPTION
**Pull request by fmeum**
Speculative fix for this crash:

```
FATAL: bazel crashed due to an internal error. Printing stack trace:
java.lang.RuntimeException: Unrecoverable error while evaluating node 'BZLMOD_REPO_RULE:@@_main~remote_jdk8_repos~remote_jdk8_macos_aarch64_toolchain_config_repo' (requested by nodes 'REPOSITORY_DIRECTORY:@@_main~remote_jdk8_repos~remote_jdk8_macos_aarch64_toolchain_config_repo')
	at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:550)
	at com.google.devtools.build.lib.concurrent.AbstractQueueVisitor$WrappedRunnable.run(AbstractQueueVisitor.java:414)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.scan(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source)
Caused by: java.lang.NullPointerException: Cannot invoke "com.google.devtools.build.lib.bazel.bzlmod.BzlmodRepoRuleValue.getRule()" because the return value of "com.google.devtools.build.lib.skyframe.BzlmodRepoRuleFunction.createRuleFromSpec(com.google.devtools.build.lib.bazel.bzlmod.RepoSpec, net.starlark.java.eval.StarlarkSemantics, com.google.devtools.build.skyframe.SkyFunction$Environment)" is null
	at com.google.devtools.build.lib.skyframe.BzlmodRepoRuleFunction.compute(BzlmodRepoRuleFunction.java:151)
	at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:461)
	... 7 more
```

### CODE REVIEW

**Comment by fmeum** [comment]
I can't trigger this reliably and the code isn't public, so no reproducer unfortunately. :-(

**Comment by iancha1992** [comment]
@bazel-io fork 7.1.0

File: src/main/java/com/google/devtools/build/lib/skyframe/BzlmodRepoRuleFunction.java [hunk 1]
```diff
@@ -148,7 +148,11 @@ public SkyValue compute(SkyKey skyKey, Environment env)
       return BzlmodRepoRuleValue.REPO_RULE_NOT_FOUND_VALUE;
     }
     RepoSpec extRepoSpec = extensionEval.getGeneratedRepoSpecs().get(internalRepo);
-    Package pkg = createRuleFromSpec(extRepoSpec, starlarkSemantics, env).getRule().getPackage();
+    BzlmodRepoRuleValue repoRuleValue = createRuleFromSpec(extRepoSpec, starlarkSemantics, env);
```
Comment (Wyverald) [inline src/main/java/com/google/devtools/build/lib/skyframe/BzlmodRepoRuleFunction.java:151]: huh. it seems that something got lost in the history here. We should actually just `return createRuleFromSpec(extRepoSpec, starlarkSemantics, env);` and be done with this :)

Comment (fmeum) [reply src/main/java/com/google/devtools/build/lib/skyframe/BzlmodRepoRuleFunction.java]: Just made that change in the browser, will see how it goes :⁠-⁠D

**Comment by iancha1992** [comment]
The changes in this PR have been included in [Bazel 7.0.1 RC2](https://releases.bazel.build/7.0.1/rc2/index.html). Please test out the release candidate and report any issues as soon as possible. If you're using Bazelisk, you can point to the latest RC by setting USE_BAZEL_VERSION=last_rc.
Thanks!

**Pull request closed without merging**